proto/executor.proto

```diff
@@ -33,13 +34,42 @@ message Executor {
 	optional TopN topN = 6;
 	optional Limit limit = 7;
 	optional Aggregation stream_agg= 8;
+	optional Join join = 9;
+	optional string executor_id = 10;
+}
+
+enum EngineType {
+	Local = 0;
+	TiKV = 1;
+	TiFlash = 2;
 }
 
 message TableScan {
 	optional int64 table_id = 1 [(gogoproto.nullable) = false];
 	repeated ColumnInfo columns = 2;
 	optional bool desc = 3 [(gogoproto.nullable) = false];
 	repeated int64 primary_column_ids = 4;
+	optional EngineType next_read_engine = 5 [(gogoproto.nullable) = false]; // which engine we should in next step, only used by tiflash
+    repeated KeyRange ranges = 6 [(gogoproto.nullable) = false];
```

Comment:
```suggestion
	repeated KeyRange ranges = 6 [(gogoproto.nullable) = false];
```

Reply:
Could you explain or document it why we need this now but do not need it before?

Reply:
For A Join B, and Table B is in global (broadcast) side. We can know the ranges of A by parsing region info, but cannot get ranges of Table B.